src/client/dfs/dfs.c

```diff
@@ -5082,13 +5090,25 @@ dfs_chmod(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode)
 	}
 
 out:
-	if (S_ISLNK(entry.mode)) {
+	if (S_ISLNK(entry.mode) && !(flag & O_NOFOLLOW)) {
 		dfs_release(sym);
 		daos_obj_close(oh, NULL);
 	}
 	return rc;
 }
 
+int
+dfs_chmod(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode)
+{
+	return chmod_int(dfs, parent, name, mode, 0);
+}
+
+int
+dfs_chmod_wflag(dfs_t *dfs, dfs_obj_t *parent, const char *name, mode_t mode, int flag)
+{
+	return chmod_int(dfs, parent, name, mode, flag);
+}
+
```

Comment:
Not sure how this affects your work, but I want to point out that chmod on a symlink itself is not always supported.
1. For example, from https://man7.org/linux/man-pages/man2/chmod.2.html:
`AT_SYMLINK_NOFOLLOW` isn't even implemented for `chmod`.

2. And this discussion: https://unix.stackexchange.com/questions/293763/how-to-change-the-symlink-permission

3. When DFS creates a symlink, it sets 777: https://github.com/daos-stack/daos/blob/d439c2fb7fa0d4a2ac39cab1791d727942b775b0/src/client/dfs/dfs.c#L1473

4. And for datamover tools I ran into this. Hence: https://github.com/daos-stack/daos/blob/d439c2fb7fa0d4a2ac39cab1791d727942b775b0/src/client/dfs/dfs_sys.c#L667-L669

Reply:
@daltonbohning You are right. "AT_SYMLINK_NOFOLLOW" was not implemented in glibc under most linux. This is OS dependent. 

@mchaarawi Since only a few systems have implemented fchmodat() and faccessat(). Rocky 8 does not support them. Shall we simply return
EOPNOTSUPP ("Operation not supported") in the interception library? If so, it may be not necessary to add dfs_chmod_wflag() and dfs_access_wflag(). Do I need to check other OS we commonly use for test?